src/internal/joiExtended.ts

```diff
@@ -126,6 +127,38 @@ export const joi = Joi
     };
   })
 
+  .extend((joi: Joi.Root) => {
+    return {
+      type: 'token',
+      base: joi.any(),
+      messages: {
+        'token.transform': 'Transformation failed for {{#label}}'
+      },
+      rules: {
+        transform: {
+          method: function(): any {
+            // @ts-ignore
+            return this.$_addRule('transform');
+          },
+          validate: function(value: any, helpers: any) {
```

Comment:
it's not really clear to me what is happening in this function. is this needed? maybe a comment to explain this